### ISSUE TITLE
semicolon is errors for statements that aren't finished i.e. with a .catch statement.  

### ISSUE THREAD

**Issue by xtianus79** [palantir/tslint#1706]
### Bug Report

- __TSLint version__: `4.0.1`
- __TypeScript version__: `2.1.1`
- __Running TSLint via__:  Node.js API

#### TypeScript code being linted

```ts
updateAllocations(combined: xxxxx[], xxxxxx: number): Observable<PersonAllocation[]> {
        this.path = '/' + personId + '/allocation';
        let body = JSON.stringify(combined);
        return this.apiService.postObs(this.path, body)
        .map((res:Response) => res.json() as xxxxxxx[])
        .catch(error); 
    }
```

with `tslint.json` configuration:

```json
"semicolon": [
        true,
        "always"
    ],
```

#### Actual behavior

src/xxxx/xxxxx/xxxxx.service.ts[23, 42]: Missing semicolon

#### Expected behavior

This should not bother me for a missing semicolon when it is not a finished method... if there is a catch at the end of the statement it should leave this alone...

**Comment by nchen63** [comment]
there's no error if you prefix your function with `function`

**Comment by adidahiya** [comment]
@xtianus79 where exactly is your lint error falling? the line number you posted exceeds the length of your code block.

**Comment by JoshuaKGoldberg** [comment]
Closing this issue for housekeeping. If this is still an issue for you @xtianus79 please do comment here and we can take another look!
